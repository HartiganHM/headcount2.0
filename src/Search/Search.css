.search {
  align-items: center;
  display: flex;
  font-size: 1.2rem;
  padding: 1.2rem; }
  .search .search-input {
    border: 1px solid grey;
    flex: 2;
    padding: 0.5rem; }
    .search .search-input:focus {
      outline: none; }
  .search .search-icon {
    height: 1.5rem;
    margin: 1rem;
    margin-left: 0;
    width: 1.5rem; }
