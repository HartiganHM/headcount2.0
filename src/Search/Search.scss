.search {
    align-items: center;
    display: flex;
    font-size: 1.2rem;
    padding: 1.2rem;

    .search-input {
        border: 1px solid grey;
        flex: 2;
        padding: 0.5rem;

        &:focus {
            outline: none;
        }
    }

    .icon-search {
        font-size: 2rem;
        margin: 1rem;
        margin-left: 0;
    }
}
