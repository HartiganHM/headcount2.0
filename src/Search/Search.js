import React, { Component } from 'react';
import icon from '../Icons/search.svg';
import PropTypes from 'prop-types';
import '../../fontello-d4ecff93/css/fontello.css';
import './Search.css';

class Search extends Component {
  constructor() {
    super();
    this.state = {
      value: ''
    };

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    let value = event.target.value;

    this.props.filterCards(value);
    this.setState({ value });
  }

  render() {
    return (
      <div className='search'>
        <i
          aria-hidden='true'
          className='icon-search'></i>
        <input
          className='search-input'
          type='text'
          placeholder='Search for a district'
          value={this.state.value}
          onChange={this.handleChange}
        />
      </div>
    );
  }
};

Search.propTypes = {
  filterCards: PropTypes.func,
};

export default Search;
