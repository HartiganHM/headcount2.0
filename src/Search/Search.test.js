import React from 'react';
import Search from './Search';
import { shallow } from 'enzyme';

describe('Search Tests', () => {
  let renderedSearch;
  let mockFunc;

  beforeEach(() => {
    mockFunc = jest.fn();
    renderedSearch = shallow(<Search filterCards={mockFunc} />);
  });

  it('Search should exist', () => {
    expect(renderedSearch).toBeDefined();
  });

  it('Search should match the snapshot', () => {
    expect(renderedSearch).toMatchSnapshot();
  });

  it('Search should fire filterCards with change', () => {
    let input = renderedSearch.find('.search-input');

    renderedSearch
      .find('.search-input')
      .simulate('change', { target: { value: 'abc' } });
    expect(mockFunc.mock.calls).toEqual([['abc']]);
  });
});
