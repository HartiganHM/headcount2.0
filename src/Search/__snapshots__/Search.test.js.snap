exports[`Search Tests Search should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    filterCards={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <i
          aria-hidden="true"
          className="icon-search" />,
        <input
          className="search-input"
          onChange={[Function]}
          placeholder="Search for a district"
          type="text" />,
      ],
      "className": "search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "className": "icon-search",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-input",
          "onChange": [Function],
          "placeholder": "Search for a district",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            aria-hidden="true"
            className="icon-search" />,
          <input
            className="search-input"
            onChange={[Function]}
            placeholder="Search for a district"
            type="text" />,
        ],
        "className": "search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "className": "icon-search",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "onChange": [Function],
            "placeholder": "Search for a district",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
